### ISSUE TITLE
Home Screen showing 0 games

### ISSUE THREAD

**Issue by pldimarco** [Moyster/BaiGfe#10]
Similar to https://github.com/Moyster/BaiGfe/issues/4. GFE showing no games on the home screen, but in the preferences, there are games found. Tried reinstalling, restarting, and clearing the cache with no luck.

On version 3.20.2.34:
![image](https://user-images.githubusercontent.com/6627983/78453029-2d2a3b80-765d-11ea-94c8-e6b16483bc60.png)

![image](https://user-images.githubusercontent.com/6627983/78453008-1257c700-765d-11ea-9e60-7e9cb8f4f828.png)

**Comment by Moyster** [comment]
That's wierd, I've been installing GFE and using the latest app.js on at least 3 different setups.  

Did you block all domains / only a few ? The most common "factor" for the game list not to show is GFE JS code missing some commas here and there, can you try uninstalling and making sure all folders are properly deleted under Program Files / Program Files (x86), or did you do that when you say "uninstalling/clearing the cache" ?

**Comment by pldimarco** [comment]
Just tried uninstalling it, removing all `Nvidia Corporation` folders from `Program Files` and `Program Files (x86)`, and reinstalling. But I hit the same result. My current blocked (checked) domains on my pihole is as follows:

- [x] telemetry.gfe.nvidia.com
- [x] gfe.nvidia.com
- [ ] gfwsl.geforce.com
- [ ] services.gfe.nvidia.com
- [ ] accounts.nvgs.nvidia.com
- [ ] events.gfe.nvidia.com
- [ ] img.nvidiagrid.net
- [x] images.nvidiagrid.net
- [ ] images.nvidia.com
- [ ] ls.dtrace.nvidia.com
- [ ] ota.nvidia.com
- [ ] rds-assets.nvidia.com
- [x] assets.nvidiagrid.net
- [ ] nvidia.tt.omtrdc.net
- [ ] api.commune.ly
- [ ] login.nvgs.nvidia.cn

**Comment by Moyster** [comment]
It might be a dumb intuition, but did you ever log in GFE ? I mean, with a real mail/account, before applying app.js ?

**Comment by pldimarco** [comment]
On my first attempt to convert if from normal use to no log-in. But haven't tried after any of the re installations. Is that necessary?

**Comment by Moyster** [comment]
I'm just wondering what can cause the game list not to show, and since a "fresh install" seems to work with the modded app.js, then maybe something goes wrong if one were previously logged before.  
I'll try to find some time (shouldn't be too hard these days) to setup a VM and do some experiment to try and reproduce this, see if there's an easy fix / workaround ;)

**Comment by pldimarco** [comment]
Yeah, and of course I was going to try and see if some files were left around. But there is version `3.20.3.63` out now. Which doesn't seem to work even partially. Let me know if you need help debugging.

**Comment by Moyster** [comment]
this shouls work for 3.20.3.63 ( https://github.com/Moyster/BaiGfe/blob/master/app.js ), what do you mean by "doesn't seem to work even partially" ?  
There was a lot of changes in GFE 3.20.3.63, older app.js might not work.

**Comment by pldimarco** [comment]
I was considering "works partially" as allows me to bypass login, but not see the games. But I had the old version of the `app.js`. So I'll try the new one.

**Comment by Moyster** [comment]
![gfesc](https://user-images.githubusercontent.com/19413599/79613272-7c14ae00-80fe-11ea-8fde-7c66f1779d7e.PNG)
Click the "detailed view" button and again to go back to "mosaic view", that's the current workaround when the gamelist doesn't show ;)

**Comment by pldimarco** [comment]
Tried it and didn't work. Even tried reinstalling, but still same behavior.

**Comment by Moyster** [comment]
Rename the app.zip to app.js (it's not zipped, just renamed for github to allow uploads), it's the latest app.js with the trick from https://github.com/Moyster/BaiGfe/issues/13  
Mostly a blind shot but I'm wondering if it helps your install get "unstuck", if you could test and report, would be nice :)
[app.zip](https://github.com/Moyster/BaiGfe/files/4497047/app.zip)

**Comment by pldimarco** [comment]
Reinstalled GFE and applied the change and the full blocklist. Now the games are showing up. No images/data. But I'm guessing that's from the full blocklist.

**Comment by Moyster** [comment]
Nice, that means i don't have to setup a VM, astrogrep, fiddler2 etc... :D  

I think I will simply delete the "lite blocklist", make a section saying something like "remove this to keep that feature" below the blocklist. If you test and do it before me, feel free to contribute! :)

**Comment by pldimarco** [comment]
Yeah. I confirmed adding the image subdomains allows the cover art to be added to the game placards. But I wasn't able to pull down the data with this whitelist. I'm assuming data refers to the game configuration settings when you click `details` for a game.

**Comment by Darthagnon** [comment]
@pldimarco Did you manage to fix your game list? I'm having mine show up as blank, too.

**Comment by pldimarco** [comment]
At first it worked. Then I whitelisted the image domains, and the images showed up. Then I tried to find the domains to unblock to get the game optimization data, and that's when the games disappeared. I haven't had a chance to reset and try again yet.

**Comment by Darthagnon** [comment]
Sounds like we did exactly the same thing

**Comment by pldimarco** [comment]
Yeah. Luckily it is easy to reset. But it feels like playing whack-a-mole with the remaining domains to find the right one.

**Comment by psisquared2** [comment]
Small heads up, an easy way to reset is to rename (or delete) the folder: 
`C:\Users\<my name>\AppData\Local\NVIDIA Corporation\NVIDIA Geforce Experience` 
Much quicker than a reinstall.
